core/src/test/java/smile/projection/PCATest.java

```diff
@@ -272,4 +279,56 @@ public void testPCACor() {
             }
         }
     }
+
```

Comment:
This is not necessary. PCA and ICA are for different purposes. And I don't see you really test any real results here.